scripts/bluetooth/mesh/mesh_dfu_metadata.py

```diff
@@ -0,0 +1,314 @@
+        self.vnd_list.append((cid << 16) + vid)
+
+    def sig_model_add(self, id):
+        self.sig_list.append(id)
+
+    def bytestring_generate(self):
+        bytestring = bytearray()
+        bytestring.extend(self.loc.to_bytes(2, 'little'))
+        bytestring.append(len(self.sig_list))
+        bytestring.append(len(self.vnd_list))
+
+        for sig in self.sig_list:
+            bytestring.extend(sig.to_bytes(2, 'little'))
+        for vnd in self.vnd_list:
+            bytestring.extend(vnd.to_bytes(4, 'little'))
+
+        return bytestring
+
+    def dict_generate(self):
+        return {
+            "location": self.loc,
+            "sig_models": self.sig_list,
+            "vendor_models": self.vnd_list,
+        }
+
+
+class Comp0:
+    # Must stay in order
+    FEATURE_KCONF_OPTS = [
+        'CONFIG_BT_MESH_RELAY',
+        'CONFIG_BT_MESH_GATT_PROXY',
+        'CONFIG_BT_MESH_FRIEND',
+        'CONFIG_BT_MESH_LOW_POWER',
+    ]
+
+    def __init__(self):
+        self.elems = []
+        self.hash = None
+        self.cid = None
+        self.pid = None
+        self.vid = None
+        self.crpl = None
+        self.feat = 0
+
+    def elem_add(self, loc):
+        new_elem = Elem(loc)
+        self.elems.append(new_elem)
+        return new_elem
+
+    def features_add(self, kconfig):
+        if 'CONFIG_BT_MESH_CRPL' not in kconfig.keys():
+            print("Could not find CONFIG_BT_MESH_CRPL Kconfig option")
+            sys.exit(1)
+
+        self.crpl = int(kconfigs['CONFIG_BT_MESH_CRPL'])
+
+        for i, opt in enumerate(self.FEATURE_KCONF_OPTS):
+            opt_set = False
+            try:
+                opt_set = kconfig[opt] == 'y'
+            except Exception:
+                pass
+
+            self.feat += (1 if opt_set else 0) << i
+
+    def __bytestring_generate(self):
+        bytestring = bytearray()
+        bytestring.extend(self.cid.to_bytes(2, 'little'))
+        bytestring.extend(self.pid.to_bytes(2, 'little'))
+        bytestring.extend(self.vid.to_bytes(2, 'little'))
+        bytestring.extend(self.crpl.to_bytes(2, 'little'))
+        bytestring.extend(self.feat.to_bytes(2, 'little'))
+
+        for elem in self.elems:
+            bytestring.extend(elem.bytestring_generate())
+
+        return bytestring
+
+    def dict_generate(self):
+        j_dict = {
+            "cid": self.cid,
+            "pid": self.pid,
+            "vid": self.vid,
+            "crpl": self.crpl,
+            "features": self.feat,
+            "elements": []
+        }
+
+        for elem in self.elems:
+            j_dict["elements"].append(elem.dict_generate())
+
+        return j_dict
+
+    def hash_generate(self):
+        # Uses 16-byte zero key
+        crypto = cmac.CMAC(algorithms.AES(bytes(16)))
+        crypto.update(bytes(self.__bytestring_generate()))
+        self.hash, *_ = struct.unpack('<L', crypto.finalize()[:4])
+        return self.hash
+
+
+def get_die_var_addr(die):
```

Comment:
I disagree with this. The class has no state (or, it doesn't need to have state, it has now because of the arbitrary split that `find_comp_data_from_dwarf` is called in `__init__` and `read_comp_data` is not.) and makes no sense as a class just to group a lot of (things that should all be) `@staticmethod`s.

My suggestion:
  * inline `get_die_var_addr`, it is used only once and with my suggested rewrite it could be one or two lines long.
  * either:
      * create a `parse_comp_data` which calls both `find_comp_data_from_dwarf` and `read_comp_data` (potentially inlining the latter)
      * The one I prefer: create `@classmethod Comp0::from_elf() -> [Comp0]` which does the same.

Reply:
Fine by me

Reply:
@PavelVPV  @Andrewpini  resolving this now, since it seems we all agree now and the class is gone.